src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = [];
+  const dateObject = {};
+  const dateValues = date.split(fromFormat.at(-1));
+
+  for (let i = 0; i < dateValues.length; i++) {
+    dateObject[fromFormat[i]] = dateValues[i];
+  }
+
+  if (fromFormat.includes('YYYY')) {
+    if (toFormat.includes('YY')) {
+      dateObject['YY'] = dateObject['YYYY'].slice(2);
+    }
+  } else if (toFormat.includes('YYYY')) {
+    dateObject['YYYY'] =
+      +dateObject['YY'] < 30
+        ? `20${dateObject['YY']}`
+        : `19${dateObject['YY']}`;
+  }
+
+  for (const name of toFormat.slice(0, -1)) {
+    newDate.push(dateObject[name]);
+  }
```

Comment:
The loop iterates over `toFormat.slice(0, -1)`, which means it excludes the last element of `toFormat`. This might not be the intended behavior if you want to include all elements in the `toFormat` array. Consider revising this logic if you want to include the last element.